test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -361,6 +361,63 @@ public class InnerAttributeRouteTest
         }
 
         [Theory]
+        [InlineData("template", "template/{parameter}", "/template/5", true)]
+        [InlineData("template", "template/{parameter}", "/template", false)]
+        [InlineData("template/{parameter}", "template/{parameter}/{id}", "/template/5/1234", true)]
+        [InlineData("template", "template/{parameter:int=1033}", "/template", false)]
+        [InlineData("template", "template/{parameter:int=5}", "/template/5", true)]
+        public void AttributeRoute_GenerateLink_UseAvailableVariables(
+            string firstTemplate,
+            string secondTemplate,
+            string expectedPath,
+            bool useValues)
+        {
+            // Arrange
+            var expectedGroup = CreateRouteGroup(0, useValues ? secondTemplate : firstTemplate);
+
+            string selectedGroup = null;
+
+            var next = new Mock<IRouter>();
+            next.Setup(n => n.GetVirtualPath(It.IsAny<VirtualPathContext>())).Callback<VirtualPathContext>(ctx =>
+            {
+                selectedGroup = (string)ctx.ProvidedValues[AttributeRouting.RouteGroupKey];
+                ctx.IsBound = true;
+            })
+            .Returns((VirtualPathData)null);
+
+            var matchingRoutes = Enumerable.Empty<AttributeRouteMatchingEntry>();
+
+            var firstEntry = CreateGenerationEntry(firstTemplate, requiredValues: null);
+            var secondEntry = CreateGenerationEntry(secondTemplate, requiredValues: null, order: 0);
```

Comment:
Do these values end up with a different `Order`? if it's the same then leave it out :+1: